### PR TITLE
Add soft_destroy_all to ActiveRecordBase.

### PR DESCRIPTION
**Pull request by kazuhisa**
I have implemented a `soft_destroy_all`.
Delete multiple rows can be easily.

### CODE REVIEW

**Comment by kazuhisa** [comment]
ping @alfa-jp

**Comment by alfa-jpn** [comment]
Sorry for the late reply.
Thank you for great PR!
I'm sorry to trouble you, but could you do a little correction?

best regards.

File: spec/kakurenbo_puti/active_record_base_spec.rb [hunk 1]
```diff
@@ -327,4 +327,38 @@
...
+      context 'When soft-deleted' do
+        it 'Return a relation without soft-deleted model.' do
+          expect {
+            model_class.soft_destroy_all(id: soft_destroyed_instance.id)
+          }.to change {
+            subject.count
+          }.by(1)
+        end
+      end
+    end
```
Comment (alfa-jpn) [inline spec/kakurenbo_puti/active_record_base_spec.rb:357]: Could you revise this test as follows?

``` ruby
describe '.soft_destroy_all' do
  subject do
    model_class.soft_destroy_all
  end
  let!(:model_instance) { model_class.create! }

  it 'Soft-Delete model' do
    ...
  end

  context 'with conditions' do
    subject do
      model_class.soft_destroy_all(id: model_instance.id)
    end

    it 'Soft-Delete model' do
     ...
    end
  end
end
```

**Comment by kazuhisa** [comment]
OK!
Please wait a few days.

**Comment by kazuhisa** [comment]
@alfa-jpn 
Fixed test. I hope that you like this correction.

File: spec/kakurenbo_puti/active_record_base_spec.rb [hunk 2]
```diff
@@ -327,4 +327,33 @@
...
+    let!(:model_instance) { model_class.create! }
+
+    it 'Soft-Delete model' do
+      expect {
+        subject
+      }.to change {
+        model_class.without_soft_destroyed.count
+      }.from(16).to(0)
+    end
```
Comment (alfa-jpn) [inline spec/kakurenbo_puti/active_record_base_spec.rb:343]: Order of Rspec test is not guaranteed.
So, this code should rewrite it as follows.

``` ruby
expect {
  subject
}.to change {
  model_class.without_soft_destroyed.count
}.to(0)
```

File: spec/kakurenbo_puti/active_record_base_spec.rb [hunk 3]
```diff
@@ -327,4 +327,33 @@
...
+      subject do
+        model_class.soft_destroy_all(id: model_instance.id)
+      end
+
+      it 'Soft-Delete model' do
+        expect {
+          subject
+        }.to change {
+          model_class.without_soft_destroyed.count
+        }.from(1).to(0)
```
Comment (alfa-jpn) [inline spec/kakurenbo_puti/active_record_base_spec.rb]: Similar.

``` ruby
expect {
  subject
}.to change {
  model_class.without_soft_destroyed.count
}.to(0)
```

**Comment by alfa-jpn** [comment]
Thank you for the speedy response.

Please fix some issues, and squash your commits into one.
https://help.github.com/articles/about-git-rebase/#rebasing-commits-against-a-point-in-time

Best regards,

**Comment by kazuhisa** [comment]
Thanks for the advice. Make sure the fix please.

**Comment by alfa-jpn** [comment]
LGTM.
Thanks for good PR!!
I release the new version in a few days.

**Pull request merged**